go-controller/pkg/node/egressip.go

```diff
@@ -0,0 +1,210 @@
+// +build linux
+
+package node
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	egressipv1 "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressip/v1"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	"github.com/vishvananda/netlink"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+	utilnet "k8s.io/utils/net"
+)
+
+const egressLabel = "egress"
+
+type egressIPLocal struct {
+	gatewayIPv4        string
+	gatewayIPv6        string
+	nodeName           string
+	defaultGatewayIntf string
+}
+
+func (n *OvnNode) watchEgressIP(egressIPLocal *egressIPLocal) error {
+	_, err := n.watchFactory.AddEgressIPHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			eIP := obj.(*egressipv1.EgressIP)
+			if err := egressIPLocal.addEgressIP(eIP); err != nil {
+				klog.Error(err)
+			}
+		},
+		UpdateFunc: func(old, new interface{}) {
+			oldEIP := old.(*egressipv1.EgressIP)
+			newEIP := new.(*egressipv1.EgressIP)
+			if err := egressIPLocal.deleteEgressIP(oldEIP); err != nil {
+				klog.Error(err)
+			}
+			if err := egressIPLocal.addEgressIP(newEIP); err != nil {
+				klog.Error(err)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			eIP := obj.(*egressipv1.EgressIP)
+			if err := egressIPLocal.deleteEgressIP(eIP); err != nil {
+				klog.Error(err)
+			}
+		},
+	}, egressIPLocal.syncEgressIPs)
+	return err
+}
+
+func (e *egressIPLocal) isNodeIP(link netlink.Link, addr *netlink.Addr) (bool, error) {
+	nodeAddrs, err := netlink.AddrList(link, netlink.FAMILY_ALL)
+	if err != nil {
+		return true, err
+	}
+	for _, nodeAddr := range nodeAddrs {
+		if nodeAddr.Label != getEgressLabel(e.defaultGatewayIntf) {
+			if addr.IP.Equal(nodeAddr.IP) {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+func (e *egressIPLocal) addHandler(link netlink.Link, egressIP string) error {
+	egressAddr, err := netlink.ParseAddr(fmt.Sprintf("%s/32", egressIP))
```

Comment:
@alexanderConstantinescu do we need to do something different for IPv6, like /128 instead? Or do we not yet support IPv6? If we don't support IPv6, then why use FAMILY_ALL for the AddrList() calls?

Reply:
Thanks for spotting that. We do support IPv6.